prow/cmd/sinker/main.go

```diff
@@ -151,6 +159,44 @@ type sinkerReconciliationMetrics struct {
 	podRemovalErrors map[string]int
 }
 
+// Prometheus Metrics
+var (
+	sinkerMetrics = struct {
+		podsCreated      prometheus.Gauge
+		timeUsed         prometheus.Gauge
+		podsRemoved      *prometheus.GaugeVec
+		podRemovalErrors *prometheus.GaugeVec
+	}{
+		podsCreated: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "pods_created",
+			Help: "Number of pods created in each sinker cleaning.",
+		}),
+		timeUsed: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "loop_duration_nanoseconds",
+			Help: "Time used in each sinker cleaning.",
+		}),
+		podsRemoved: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "pods_removed",
+			Help: "Number of pods removed in each sinker cleaning.",
+		}, []string{
+			"reason",
+		}),
+		podRemovalErrors: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "pod_removal_errors",
+			Help: "Number of errors which occurred in each sinker cleaning.",
+		}, []string{
+			"content",
```

Comment:
Right now the error messages that the k8s client will get will include the name of the Pod that is being deleted, so every message will be unique -- we should look into the k8s client helpers for deciding the reason. So, the "content" field here could be "reason" and use the `StatusReason` constants in `k8s.io/apimachinery/pkg/api/meta/v1/types.go` -- we can get that value by using `ReasonForError()` in `k8s.io/apimachinery/pkg/api/errors/errors.go`

Reply:
Done.